pymongo/pool.py

```diff
@@ -1377,7 +1373,8 @@ def __init__(self, address: _Address, options: PoolOptions, handshake: bool = Tr
         # LIFO pool. Sockets are ordered on idle time. Sockets claimed
         # and returned to pool from the left side. Stale sockets removed
         # from the right side.
-        self.conns: collections.deque = collections.deque()
+        self.available_conns: collections.deque = collections.deque()
+        self.active_conns: list[Connection] = []
         self.lock = _create_lock()
         self.active_sockets = 0
```

Comment:
There's now an ambiguity between `active_sockets` and active_conns`. Is it the case that active_sockets == len(self.active_conns)? Even if there is a reason to keep both,  it would be a good exercise to review where each is used and make sure they behave as expected.

Reply:
I believe both are required, as `active_sockets` tracks socket creation and deletion, which is not coupled exactly with connection check-in and check-out, which `active_conns` tracks.